src/sentry/event_manager.py

```diff
@@ -688,6 +692,53 @@ def _pull_out_data(jobs, projects):
         )
 
 
+def _is_commit_sha(version: str):
+    return re.match(r"[0-9a-f]{40}", version) is not None
+
+
+def _associate_commits_with_release(release: Release, project: Project):
+    previous_release = release.get_previous_release(project)
+    possible_repos = (
```

Comment:
Why do we have multiple potential repos associated to a project?

Reply:
I don't think we stop multi repos from being associated to a project. Commits should still be unique between repos.